morepath/directive.py

```diff
@@ -273,19 +275,19 @@ def predicate_key(self, reg_registry, predicate_registry):
         # dispatch_external_predicates functions that have been used,
         # or we should only allow their registration through a special
         # Morepath directive so that we can.
-        if self.func.external_predicates:
+        actual = reg_registry[self.func]
+        if actual.external_predicates:
             if not predicate_registry.get_predicates(self.func):
-                reg_registry.register_external_predicates(self.func, [])
-        reg_registry.register_dispatch(self.func)
-        return reg_registry.key_dict_to_predicate_key(
-            self.func.wrapped_func, self.key_dict)
+                actual.register_external_predicates([])
+#        reg_registry.register_dispatch(actual)
+        return actual.key_dict_to_predicate_key(self.key_dict)
```

Comment:
Note that on the version of the Reg decoupling branch I made this is allowed to become very simple without any registration happening during identifier work at all, namely:

```
     return (self.func.wrapped_func,
                self.func.key_dict_to_predicate_key(self.key_dict))
```

That fixes a longstanding issue of weird registration during identifier registration.